Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -855,4 +890,10 @@ public class MergedSQLiteBookmarks: BookmarksModelFactorySource {
         self.local = SQLiteBookmarks(db: db)
         self.buffer = SQLiteBookmarkBufferStorage(db: db)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        return modelFactory >>== {
+            return ($0 as! KeywordSearchSource).getURLForKeywordSearch(keyword)
```

Comment:
I suppose I should rephrase this as "keyword searching is nothing to do with Model".

`BookmarksModel` exists for one purpose: to back a hierarchical view of bookmark folders.

Other things about bookmarks live in other protocols.